yt_dlp/extractor/bluesky.py

```diff
@@ -0,0 +1,379 @@
+        # app.bsky.embed.recordWithMedia.view
+        entries.extend(self._extract_videos(
+            post, video_id, embed_path=('embed', 'media'), record_subpath=('embed', 'media')))
+        # app.bsky.embed.record.view
+        if nested_post := traverse_obj(post, ('embed', 'record', ('record', None), {dict}, any)):
+            entries.extend(self._extract_videos(
+                nested_post, video_id, embed_path=('embeds', 0), record_path='value'))
+        if len(entries) == 0:
+            self.raise_no_formats('No video could be found in this post', expected=True)
+        if len(entries) == 1:
+            return entries[0]
+        return self.playlist_result(entries, video_id)
+
+    @staticmethod
+    def _build_profile_url(path):
+        return format_field(path, None, 'https://bsky.app/profile/%s', default=None)
+
+    def _extract_videos(self, root, video_id, embed_path='embed', record_path='record', record_subpath='embed'):
+        embed_path = variadic(embed_path, (str, bytes, dict, set))
+        record_path = variadic(record_path, (str, bytes, dict, set))
+        record_subpath = variadic(record_subpath, (str, bytes, dict, set))
+        entries = []
+        if external_uri := traverse_obj(root, (((*record_path, *record_subpath), embed_path), 'external', 'uri', {url_or_none}, any)):
+            entries.append(self.url_result(external_uri))
+        if playlist := traverse_obj(root, (*embed_path, 'playlist', {url_or_none})):
+            formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+                playlist, video_id, 'mp4', m3u8_id='hls', fatal=False)
+        else:
+            return entries
+        video_cid = traverse_obj(
+            root, (*embed_path, 'cid', {str}),
+            (*record_path, *record_subpath, 'video', 'ref', '$link', {str}))
+        did = traverse_obj(root, ('author', 'did', {str}))
+        if did and video_cid:
+            endpoint = self.get_service_endpoint(did, video_id)
+            formats.append({
+                'format_id': 'blob',
+                'url': update_url_query(self._BLOB_URL % endpoint, {'did': did, 'cid': video_cid}),
+                **traverse_obj(root, (*embed_path, 'aspectRatio', {
+                    'width': ('width', {int_or_none}),
+                    'height': ('height', {int_or_none}),
+                })),
+                **traverse_obj(root, (*record_path, *record_subpath, 'video', {
+                    'filesize': ('size', {int_or_none}),
+                    'ext': ('mimeType', {mimetype2ext}),
+                })),
+            })
+            for sub_data in traverse_obj(root, (
+                    *record_path, *record_subpath, 'captions', lambda _, v: v['file']['ref']['$link'])):
+                subtitles.setdefault(sub_data.get('lang') or 'und', []).append({
+                    'url': update_url_query(self._BLOB_URL % endpoint, {'did': did, 'cid': sub_data['file']['ref']['$link']}),
```

Comment:
line was too long before the `_BLOB_URL_TMPL` name change
```suggestion
                    'url': update_url_query(
                        self._BLOB_URL_TMPL % endpoint, {'did': did, 'cid': sub_data['file']['ref']['$link']}),
```